cmd/sanssh/client/client.go

```diff
@@ -267,6 +257,61 @@ func Run(ctx context.Context, rs RunState) {
 	ctx, cancel := context.WithTimeout(ctx, rs.Timeout)
 	defer cancel()
 
+	exitCode := subcommands.ExitSuccess
+
+	// If there's no batch size set then it'll be the whole thing so we can use one loop below.
+	if rs.BatchSize == 0 {
+		rs.BatchSize = len(rs.Targets)
+	}
+	// Save original lists since we'll replace rs with sub slices
+	output := state.Out
+	errors := state.Err
+
+	batchCnt := 0
+	if len(rs.Targets) > 0 {
+		batchCnt = len(rs.Targets) / rs.BatchSize
+	}
+	// How many batches? Integer math truncates so we have to do one more after for remainder.
+	for i := 0; i < batchCnt; i++ {
+		// Set up a connection to the sansshell-server (possibly via proxy).
+		conn, err := proxy.Dial(rs.Proxy, rs.Targets[i*rs.BatchSize:rs.BatchSize*(i+1)], ops...)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Could not connect to proxy %q node(s) in batch %d: %v\n", rs.Proxy, i, err)
+			os.Exit(1)
+		}
+		defer func() {
+			if err := conn.Close(); err != nil {
```

Comment:
probably not a big deal, but defer is function scoped, so you won't close any connections until you've processed all batches and exit Run(). (We already have this with open files, but in this case, you're consuming targets/batchSize proxy conns, so it _could_ be meangingful in the case of many targets and a tiny batch size)

in this case, the scope of conn is pretty small, and there are no early returns, so you could probably just explicitly close it after line 292 ( although someone could later introduce a bug with an early exit before then).

Up to you.

Reply:
True. Just closed inline.